ansible/roles/test/files/ptftests/inner_hash_test.py

```diff
@@ -0,0 +1,505 @@
+                scapy.IPv6(src=ipv6_src, dst=ipv6_dst, fl=ipv6_fl, tc=ipv6_tc, hlim=ipv6_hlim, nh=47)/ \
+                nvgre_hdr
+
+        if inner_frame:
+            pkt = pkt / inner_frame
+        else:
+            pkt = pkt / scapy.IP()
+            pkt = pkt/("D" * (pktlen - len(pkt)))
+
+        return pkt
+
+
+    def check_ipv6_route_nvgre(self, hash_key, src_port, ip_dst, ip_src, dport, sport, ip_proto):
+        '''
+        @summary: Check IPv6 nvgre based inner packet hashing works
+        @param hash_key: hash_key we are varying in this iteration
+        @param src_port: index of port to use for sending packet to switch
+        @param ip_dst: Inner packet Destination IP address
+        @param ip_src: Inner packet Source IP address
+        @param dport: Inner packet dst port
+        @param sport: Inner packet src port
+        @param ip_proto: Inner packet ip protocol
+        '''
+        src_mac = self.dataplane.get_mac(0, src_port)
+
+        outer_ip_src = self.outer_src_ip_interval.get_random_ip() if hash_key == 'outer-tuples' else self.outer_src_ip_interval.get_first_ip()
+        outer_ip_dst = self.outer_dst_ip_interval.get_random_ip() if hash_key == 'outer-tuples' else self.outer_dst_ip_interval.get_first_ip()
+
+        pkt = self.generate_inner_pkt(sport, dport, ip_src, ip_dst, ip_proto)
+        nvgre_pkt = self.simple_nvgrev6_packet(
+                    eth_dst=self.router_mac,
+                    eth_src=src_mac,
+                    ipv6_src=outer_ip_src,
+                    ipv6_dst=outer_ip_dst,
+                    nvgre_tni=random.randint(1, 254)+20000,
+                    inner_frame=pkt)
+
+        logging.info("Sending packet from port {} outer_ip_src {} outer_ip_dst {} inner_src_ip {} inner_dst_ip {} inner_sport {} inner_dport {} inner_ipproto {}"\
+                .format(src_port, outer_ip_src, outer_ip_dst, ip_src, ip_dst, sport, dport, ip_proto))
+        send_packet(self, src_port, nvgre_pkt)
+
+        masked_exp_pkt = Mask(nvgre_pkt)
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "src")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IPv6, "hlim")
+
+        return verify_packet_any_port(self, masked_exp_pkt, self.exp_port_list)
+
+
+    def check_ip_route_vxlan(self, hash_key, src_port, ip_dst, ip_src, dport, sport, ip_proto):
+        '''
+        @summary: Check IP vxlan based inner packet hashing works
+        @param hash_key: hash_key we are varying in this iteration
+        @param src_port: index of port to use for sending packet to switch
+        @param ip_dst: Inner packet Destination IP address
+        @param ip_src: Inner packet Source IP address
+        @param dport: Inner packet dst port
+        @param sport: Inner packet src port
+        @param ip_proto: Inner packet ip protocol
+        '''
+        if ip_network(unicode(self.outer_dst_ip)).version == 4:
```

Comment:
@anish-n suggest to add a named definition for address type, since native python library doesn't have it